castanet/counter.py

```diff
@@ -0,0 +1,121 @@
+        imports_dictionary[file] = len(imports_list)
+
+    return imports_dictionary
+
+
+def match_comment(cast_dict):
+    """A function for counting the number of comments statements in a Python program."""
+    comment_dictionary = {}
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of comments for each file
+        comment = match.findall(cast, match.Comment())
+        comment_dictionary[file] = len(comment)
+
+    return comment_dictionary
+
+
+def count_whileloops(cast_dict):
+    """A function for counting the number of while loops in a Python program."""
+    while_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        while_loops = match.findall(cast, match.While())
+        while_dictionary[file] = len(while_loops)
+
+    return while_dictionary
+
+
+def count_forloops(cast_dict):
+    """A function for counting the number of while loops in a Python program."""
+    for_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of for statements for each file
+        imports = match.findall(cast, match.For())
+        for_dictionary[file] = len(imports)
+
+    return for_dictionary
+
+
+def match_if_statements(cast_dict):
+    """A function for counting the number of if statements in a Python program."""
+    if_statements_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of comments for each file
+        ifs = match.findall(cast, match.If())
+        if_statements_dictionary[file] = len(ifs)
+
+    return if_statements_dictionary
+
+
+def match_funcdefs(cast_dict):
+    """A function for counting the number of function definitions in a Python program."""
+    func_count = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # track the number of docstrings
+        docstring_num = 0
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of import statements for each file
+        funcdefs = match.findall(cast, match.FunctionDef())
+        # store the number of functions
+        func_count[file] = {"function" : 0, "docstring": 0}
+        func_count[file]["function"] = len(funcdefs)
+        # iterate and count the number of docstrings
+        for node in funcdefs:
+            if node.get_docstring():
+                docstring_num += 1
+        func_count[file]["docstring"] = docstring_num
+
+    return func_count
+
+
+def count_function_doc(func_count: Dict) -> int:
```

Comment:
This name could be more descriptive; it doesn't reflect that it counts functions without docstrings.

Reply:
This suggestion can be completed next sprint.